runtime/src/iree/base/loop_emscripten.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This is the JavaScript side of loop_emscripten.c
+//
+// References:
+//   * https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html
+//   * https://github.com/evanw/emscripten-library-generator
+//   * https://github.com/emscripten-core/emscripten/tree/main/src
+
+const LibraryLoopEmscripten = {
+  $loop_emscripten_support__postset: 'loop_emscripten_support();',
+  $loop_emscripten_support: function() {
+    class LoopEmscripten {
+      constructor() {
+        // TODO(scotttodd): store state here
+      }
+
+      loop_command_call(callback, user_data, loop) {
+        console.log('loop_command_call');
+
+        const iree_ok_status = 0;
+
+        setTimeout(() => {
+          console.log('loop_command_call -> timeout, calling function');
+          const ret =
+              Module['dynCall_iiii'](callback, user_data, loop, iree_ok_status);
+          console.log('function result:', ret);
+          // TODO(scotttodd): handle the returned status (sticky failure state?)
+        }, 0);
+
+        return iree_ok_status;
+      }
+    }
+
+    const instance = new LoopEmscripten();
```

Comment:
if you start storing state on this side you'll want to have a way to identify the loop so that you don't cross the streams. you can call out to here in `iree_loop_emscripten_allocate` to create the LoopEmscripten, stash it in an array, and then return the index to C to store on its struct - you'd then pass that index out each time you went C->JS, and that then also lets you also call out in `iree_loop_emscripten_free` to remove it from the array and do any JS-side cleanup you need (aborting promises, cleaning up pending callback maps, etc).

for now since this is stateless it's probably fine, but if not now you'll want to do this next.

Reply:
Yep, that's next on my list, and the intent behind the TODOs like
```
  // TODO(scotttodd): handle to a "scope"/object (managed in JS), so multiple
  //                  loops can exist at once
```